src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -142,7 +146,7 @@ class PacketDispatcher : public packet::IWriter {
         }
     }
 
-    bool pop_source() {
+    bool push_one_source() {
```

Comment:
push_written() pushes all written source and repair packets.

Accordingly, I would assume that push_source() pushes all source packets, but it actually pushes exactly one packet.

So we have three methods:

* push_written (push all written source and repair packets)
* push_one_source (push one written source packet)
* push_delayed (push one delayed packet)

Suggestions?

Also, I'd rather rename them *after* merging two my two PRs because there will be many conflicts.

Reply:
`push_written` is about pushing all packets.
`push_delayed` is about pushing one delayed packet.

From this point of view I would expect that `push_foo` will push all packets on some kind.

I would suggest the following names:

* push_delayed (push one delayed packet)
* push_source (push one source packet)
* drain (empty stock queues), probably the name a little bit awkward :)

Reply:
It would be surprising that push_source() is a special case of drain(). I'd expect all of them to be named similarly.

Reply:
Another idea:

* push_source_stock(int limit) (push at most limit packets from source stock)
* push_stocks() (push all packets from source and repair stocks)
* push_delayed(int index) (push delayed packet)

Reply:
I like your approach. Lets rename them after #192.